File: client/src/app/core/entity/grnproduct.ts
import {Product} from "./product";
import {Grn} from "./grn";

export class GrnProduct{

  id: number;
  product:Product;
  quentity:number;

  constructor(id:number,product:Product,quentity:number) {
    this.id = id;
    this.product = product;
    this.quentity = quentity;
  }

}

File: server/src/main/java/com/content/security/entity/Grnproduct.java
package com.content.security.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "grnproduct")
public class Grnproduct {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "grn_id", nullable = false)
    private Grn grn;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quentity")
    private Integer quentity;

}